From: Sven Eckelmann <sven@narfation.org>
Date: Tue, 10 Dec 2024 22:31:33 +0100
Subject: Revert "batman-adv: move asm/unaligned.h to linux/unaligned.h"

This header is only available with Linux 6.12

This reverts commit ee60832ebec47a023d634b06f9434103ec090aed.

--- a/compat-include/linux/unaligned.h
+++ /dev/null
@@ -1,20 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 */
-/* Copyright (C) B.A.T.M.A.N. contributors:
- *
- * Marek Lindner, Simon Wunderlich
- *
- * This file contains macros for maintaining compatibility with older versions
- * of the Linux kernel.
- */
-
-#ifndef _NET_BATMAN_ADV_COMPAT_LINUX_UNALIGNED_H_
-#define _NET_BATMAN_ADV_COMPAT_LINUX_UNALIGNED_H_
-
-#include <linux/version.h>
-#if LINUX_VERSION_IS_GEQ(6, 12, 0)
-#include_next <linux/unaligned.h>
-#else
-#include <asm/unaligned.h>
-#endif
-
-#endif /* _NET_BATMAN_ADV_COMPAT_LINUX_UNALIGNED_H_ */
--- a/net/batman-adv/distributed-arp-table.c
+++ b/net/batman-adv/distributed-arp-table.c
@@ -7,7 +7,7 @@
 #include "distributed-arp-table.h"
 #include "main.h"
 
-#include <linux/unaligned.h>
+#include <asm/unaligned.h>
 #include <linux/atomic.h>
 #include <linux/bitops.h>
 #include <linux/byteorder/generic.h>
